server/src/main/java/org/elasticsearch/index/mapper/TimeSeriesIdFieldMapper.java

```diff
@@ -142,6 +144,15 @@ public void postParse(DocumentParserContext context) throws IOException {
                 : null,
             timeSeriesId
         );
+
+        // We need to add the uid or id to nested Lucene documents so that when a document gets deleted, the nested documents are
+        // also deleted. Usually this happens when the nested document is created (in DocumentParserContext#createNestedContext), but
+        // for time-series indices the _id isn't available at that point.
+        var binaryId = context.doc().getField(IdFieldMapper.NAME).binaryValue();
```

Comment:
`getField` is kind of expensive since it iterates over all fields. Let's do this only when there are non root documents. Or maybe we can return the id from `TsidExtractingIdFieldMapper` above.